archive/find_test.go

```diff
@@ -0,0 +1,97 @@
+package archive
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/brimsec/zq/pkg/test"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zio/detector"
+	"github.com/brimsec/zq/zio/tzngio"
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/stretchr/testify/require"
+)
+
+func createArchiveSpace(t *testing.T, datapath string, srcfile string, co *CreateOptions) {
+	ark, err := CreateOrOpenArchive(datapath, co, nil)
+	require.NoError(t, err)
+
+	zctx := resolver.NewContext()
+	reader, err := detector.OpenFile(zctx, srcfile, detector.OpenConfig{})
+	require.NoError(t, err)
+	defer reader.Close()
+
+	err = Import(context.Background(), ark, reader)
+	require.NoError(t, err)
+}
+
+func indexArchiveSpace(t *testing.T, datapath string, ruledef string) {
+	rule, err := NewRule(ruledef)
+	require.NoError(t, err)
+
+	ark, err := OpenArchive(datapath, nil)
+	require.NoError(t, err)
+
+	err = IndexDirTree(ark, []Rule{*rule}, "_", nil)
+	require.NoError(t, err)
+}
+
+func findQuery(t *testing.T, ark *Archive, query IndexQuery, opts ...FindOption) string {
+	rc, err := FindReadCloser(context.Background(), ark, query, opts...)
+	require.NoError(t, err)
+	defer rc.Close()
+
+	buf := bytes.NewBuffer(nil)
+	w := zbuf.NopFlusher(tzngio.NewWriter(buf))
+	require.NoError(t, zbuf.Copy(w, rc))
+
+	return buf.String()
+}
+
+func TestOpenOptions(t *testing.T) {
+	datapath, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(datapath)
+
+	thresh := int64(1000)
+	createArchiveSpace(t, datapath, "../tests/suite/zdx/babble.tzng", &CreateOptions{
+		LogSizeThreshold: &thresh,
+	})
+
+	_, err = OpenArchive(datapath, &OpenOptions{
+		LogFilter: []string{"foo"},
+	})
+	require.Error(t, err)
+	require.Regexp(t, "no spans", err.Error())
```

Comment:
```suggestion
	require.EqualError(t, err, "no spans")
```

Reply:
the actual error text here is longer, so would just keep the regex